File: app/src/main/java/io/sweers/catchup/model/medium/Post.java
package io.sweers.catchup.model.medium;

import android.support.annotation.NonNull;

import com.google.auto.value.AutoValue;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

@AutoValue
public abstract class Post {
  public static JsonAdapter<Post> jsonAdapter(@NonNull Moshi moshi) {
    return new AutoValue_Post.MoshiJsonAdapter(moshi);
  }

  public abstract long createdAt();

  public abstract String creatorId();

  public abstract String homeCollectionId();

  public abstract String id();

  public abstract String title();

  public abstract String uniqueSlug();

  public abstract Virtuals virtuals();
}

File: services/medium/src/main/kotlin/io/sweers/catchup/service/medium/model/Post.kt
/*
 * Copyright (C) 2019. Zac Sweers
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.sweers.catchup.service.medium.model

import com.squareup.moshi.JsonClass
import java.time.Instant

@JsonClass(generateAdapter = true)
internal data class Post(
  val createdAt: Instant,
  val creatorId: String,
  val homeCollectionId: String,
  val id: String,
  val title: String,
  val uniqueSlug: String,
  val virtuals: Virtuals,
  val isSubscriptionLocked: Boolean = false
)
